interface/data_types.py

```diff
@@ -0,0 +1,57 @@
+"""
+This module defines public library data containers and type aliases.
+See test/sample_app.py for example usage.
+"""
+
+from __future__ import annotations
+from typing import NamedTuple
+
+
+class OutputData(NamedTuple):
```

Comment:
given that this has very nasty confusable arguments (string, string, int, string, ....)
I strongly vote against named tuple. It should be data class instead

Reply:
I've read a few articles and can't figure out what would be the difference. Can you elaborate, please?

Reply:
Sure. Namedtuple *can* be destructured using tuple destructuring. Dataclass cannot. This prevents you from accidentally shooting yourself to foot as you always must access through named parameters